lib/spack/spack/version.py

```diff
@@ -486,7 +485,8 @@ def satisfies(self, other):
         return (self.overlaps(other) or
                 # if either self.start or other.end are None, then this can't
                 # satisfy, or overlaps() would've taken care of it.
-                self.start and other.end and self.start.satisfies(other.end))
+                self.start is not None and other.end is not None and
```

Comment:
I haven't yet found a way to get the previous code to error here, but it seems clear we are checking for None. If anyone can figure out how this could fail, I'll add a unit test.